### ISSUE TITLE
Steam Deck controller buttons no longer functioning properly

### ISSUE THREAD

**Issue by RootCellar** [ValveSoftware/SteamOS#1678]
#### Your system information

* Steam client version:1726683985
* SteamOS version: 3.6.19
* Opted into Steam client beta?: [Yes/No] No
* Opted into SteamOS beta?: [Yes/No] No
* Have you checked for updates in Settings > System?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen.

When I use the STEAM and "..." buttons on my Steam Deck, they no longer work correctly after updating to 3.6.19 on stable. The buttons don't seem to do anything at all.
The left joystick doesn't navigate game mode menus anymore either, and the mouse appears if I move my finger across the right trackpad, which did not happen before as far as I know. Pressing the D-Pad however will select the nearest GUI item and I can use it to navigate menus. A and B work as well.

If I select a game and go to the controller settings, it tells me that there is no controller connected. While there is no external controller connected, it should still show me the regular menu for configuring controls for a game using the Steam Deck's buttons.

The controls work properly if I hold "..." while booting, and choose to boot the previous version of SteamOS.

I have submitted a system report and my steam id is `76561198859974672`

#### Steps for reproducing this issue:

1. Turn on Steam Deck, boot it as normal
2. Try to use the STEAM and "..." buttons and notice that they do not work.

**Comment by rostbacke** [comment]
Same problem here. Steam settings tell "No controller connected". But dpad works for me and as soon as I've touched the touchscreen A and B also Work. But A doesn't work inside virtual keyboard.

Decky Loader is installed with several plugins and CryUtilities with all recommended settings enabled. 
Thought the culprit was Steam-Deck.Auto-Disable-Steam-Controller (https://github.com/scawp/Steam-Deck.Auto-Disable-Steam-Controller) which I needed for several games in docked mode. But uninstallation and new update installation didn't work. 

Beta works, but Decky Loader is disabled there.

**Comment by rostbacke** [comment]
Nop

> @rostbacke can you get the Steam Deck virtual keyboard to appear in gaming mode with STEAM+X? I can't, the STEAM button does _absolutely nothing_ for me.

Nope, both the Steam button and the [...] button don't work at all after the update. And since the Steam button doesn't work, all the combinations won't work. Neither in gaming mode, nor in desktop mode. 

Had to connect a keyboard to try uninstallation of plugins I suspected to be the reason. Didn't work. Since not everybody seems to have this problem, I guess it's some sort of plugin that's causing the problems.

**Comment by RootCellar** [comment]
@rostbacke I don't think it is a plugin, as I don't have any on my Steam Deck. The only things I've done with my Steam Deck are 1) enabling `sshd`, so I can login remotely from other Linux systems and run commands, 2) a custom Zram swap, and 3) A custom distrobox install from before SteamOS had it, none of which should cause the problem because they're unrelated and didn't cause any issues in any update before.

**Comment by rostbacke** [comment]
So I think we can exclude Decky Loader and its plugins as the culprit.
Do you have an LCD or OLED deck? Looking at reddit and the update comments, we're not the only ones with the problem.

**Comment by RootCellar** [comment]
LCD. Older one that I got at the end of April, 2022.

**Comment by rostbacke** [comment]
Same here. Haven't seen an OLED owner complaining about the controller issue.

**Comment by matte-schwartz** [comment]
Would recommend anyone experiencing the issue to also submit a system report as described in https://github.com/ValveSoftware/SteamOS?tab=readme-ov-file#reporting-issues since it will contain info that the SteamOS maintainers should find useful

**Comment by rostbacke** [comment]
System report sent with Steam ID 76561198129037992

**Comment by deathblade201243** [comment]
oddly it seems to be only the Stable version of 3.6.19 that has the issue (among others issues) which the Beta version of 3.6.19 doesn't have. idk what has changed between the two though.

**Comment by matte-schwartz** [comment]
that's actually interesting... from what I can see, there only seems to be one version of the SteamOS image (20241021.100) on both Stable and Beta branches right now. I think this means the only difference would have to come from the Steam client itself rather than SteamOS. 

The Stable option puts you on the stable branch for both the SteamOS image (20241021.100) and Steam client version (1726683985), while Preview puts you on the beta branch for the SteamOS image (which is also 20241021.100) and Steam cleint version (1729208001).

**Comment by Eduxki** [comment]
Cool, a useless device because some genius decided to download without consent an automatic conttoller stop for peppe dumb enough to use a steam deck as a pc

**Comment by Eduxki** [comment]
Ok beta seems to work, fuk sloppy maintainers of semi functional plugins that lick the device whole thebtime

**Comment by SongTaeHa** [comment]
Hey.

I had the same problem and it stopped working with 3.6.19 on stable channel.

Beta worked for me and controller issues are fixed.

But to add, in the past I've tried previously to hop on preview channel multiple times with 3.6.x steam deck and every time I had these controller issues. Didn't have these problem when I was using preview channel of 3.5.x right after the purchase.

**Comment by SpaceUnicornWizard** [comment]
I have the same problem with OLED model.  My problem with controller started before I switched to 3.6.19. I have received an update for a Steam Client, and after installing it, the controller stopped working, while i was still on 3.5.19. 
Because I didn't receive update to 3.6.19, I decided to switch to beta channel, and there everything worked, with tons of decky plugins, both on stable and pre-release version of decky loader. After that I have tried to switch back to stable, and controller stopped working again, both built-in and external controller connected by wire. So I switched back on beta and tried installing stable 2 more times. 
On the last attempt controller started working, but only after putting Steam Deck to sleep and waking it again. In desktop mode, the controller and touchpads work right until Steam Client starts working. Because of this I think the Steam Client is the culprit. Even lsusb in terminal sees the steam controller attached, while Steam Client says otherwise. 
So launching Steam Client, both from booting the device or switching between desktop mode and game mode seems to forget some code that allows Client to connect any controller (built-in or external) and putting to sleep and waking seems to fix this problem, in both modes.

**Comment by RootCellar** [comment]
@SpaceUnicornWizard I tried to see what would happen if I put mine to sleep and wake it, and the controls still do not work. Mine's an LCD though, as I said previously.

**Comment by rostbacke** [comment]
Sleep-Wake doesn't work for me either on LCD.

**Comment by rostbacke** [comment]
Oh, and btw, 3.6.19 cannot automatically login anymore. I always have to relogin at restart.

**Comment by RootCellar** [comment]
@rostbacke I think that's a different issue, I don't have that problem and it probably isn't related to the controller buttons not working. You may need to seek help elsewhere or submit a separate issue.

**Comment by nl255** [comment]
For me, the controls wouldn't work on boot unless I put it to sleep and then woke it up and even then they wouldn't work if I switched into desktop mode as well as hanging on "shutting down steam" when trying to reboot or shutdown.  Switching to beta fixed it though.

**Comment by kisak-valve** [comment]
Internal deck controls not working after 3.6.19 update.

Issue transferred from https://github.com/ValveSoftware/SteamOS/issues/1683.
@BlackStargenX posted on 2024-10-26T04:02:09:

#### Your system information

* Steam client version:
* SteamOS version: 3.6.19
* Opted into Steam client beta?: [No] 
* Opted into SteamOS beta?: [No] 
* Have you checked for updates in Settings > System?: [Yes]

#### Please describe your issue in as much detail as possible:
ControllerID:MHDA3071358
LCD steam deck.
SteamID:76561198247820298

After the 3.6.19 my internal face control will not work. Including joysticks and trackpads.
Their not broken as downgrading the os or switching to the beta channel brings back their functionality.
I also tried looking at the Jupiter info
"path": "/dev/hidraw3",

"vendor_id": 10462, "product_id":4613,

"serial_number": "",

"release_number": 768,

"manufacturer_string": "Valve Software", "product_string": "Steam Controller",

"usage page": 65535,

"usage": 1,

"Interface_number": 2,

"build timestamp": 1709502893,

"secondary_build_timestamp": 0,

"is_bootloader": false

Don't know if this will help.

I had reimage the system again, this time I connected to the Internet before it finished it's stepup install and first booted up in OS 3.6.19. controls are working. I also rebooted the system and their still working.  

I'm gonna go through regular steps I usually do and install proton GE and decky loader to see if anything happens. Will report if controls brake again.

#### Steps for reproducing this issue:

1. Updated to 3.6.19 from an older version.
2. 
3. 
***
@BlackStargenX commented on 2024-10-26T04:12:10:

I think I've might have found the problem.it might be proton tkg fault. I always installed proton GE and tkg for the games that runs great with them. 
But after installed them my internal controls stopped working again. So I started to uninstall them one at time.
Did tkg first loaded back in game mode and internal controls started working again.

So for some reason proton tkg and steam OS 3.6.19 doesn't like each other. Weird bug.
***
@BlackStargenX commented on 2024-10-26T04:20:32:

Welp I celebrated too soon. The steam and ••• buttons are still not working. I'll restart the desk and test. If it still doesn't work I'll uninstall  proton GE and try again. Will report if I get those two buttons working again.
***
@BlackStargenX commented on 2024-10-26T04:26:53:

Restarted the deck proton GE still installed and all buttons and track pads are working.
So my guess of either proton tkg or manually adding a custom proton version somehow is causing this bug in the LCD steam decks on this new steam OS version.
***
@matte-schwartz commented on 2024-10-26T05:28:30:

@kisak-valve this is a duplicate of https://github.com/ValveSoftware/SteamOS/issues/1678 but adding proton-tkg is the missing piece here.

1. Start on the SteamOS Stable image + client
2. Switch to desktop
3. Install protonplus as a flatpak
4. Install the latest version of proton-tkg with protonplus
5. Switch back into Game Mode

I can now repro this issue as well. You don't need to use the proton version for any games, it just has to exist as a compatibility tool it seems. More specifically, it seems like something in the proton file itself for proton-tkg.

SteamOS System Report after steps 1 -> 5 (and some earlier debugging):
[systemreport-20241026052549.txt](https://github.com/user-attachments/files/17529121/systemreport-20241026052549.txt)

Steam logs, since this is looking to maybe be a client issue of some sort:
[logs.tar.gz](https://github.com/user-attachments/files/17529122/logs.tar.gz)

***
@rostbacke commented on 2024-10-26T08:28:39:

I can also reproduce this. After uninstallation of Proton-tkg the controller is recognized again.
***
@nl255 commented on 2024-10-26T12:01:30:

I have found switching to the beta channel fixes it.  Also I was unable to shut down normally (it would hang on "shutting down steam") but again the beta channel fixed it.

**Comment by matte-schwartz** [comment]
seems like you can trigger the issue with any native-built proton version on the Steam Deck Stable client branch, it doesn't have to be proton-tkg. controller works again when switching back to beta channel. i guess there's a change that's still only on the beta branch but not on stable yet that fixes the real issue here

**Comment by austinp-valve** [comment]
Can some of you folks with the issue post the contents of your ~/.steam/steam/logs folder? In particular the controller.txt and controller_ui.txt logs

**Comment by RootCellar** [comment]
@austinp-valve 

`controller.txt`: https://pastebin.com/LRzyVBuB
`controller_ui.txt`: https://pastebin.com/7TMyKrFD

**Comment by matte-schwartz** [comment]
logs captured after booting into game mode on stable steamos + stable deck client -> switch to desktop -> install proton-tkg -> switch back to game mode (controls lost at this point):
[deck-no-controls-debug.tar.gz](https://github.com/user-attachments/files/17552065/deck-controls-debug.tar.gz)

then i selected steam deck beta client and hit yes when the steam client asked to restart itself, but it hung on "shutting down steam" for ~2 minutes before i force rebooted -> deck beta client installed itself -> game mode launches (with working controls this time):
[deck-beta-controls-debug.tar.gz](https://github.com/user-attachments/files/17552102/deck-beta-controls-debug.tar.gz)

**Comment by eshcheglov** [comment]
`vol -` and `...` hack don't work in the KDE session or in external (non-steam) applications.

Known information:

- /dev/input/js0 is present and is recognized as a valid joystick device.
- The jstest tool from the joyutils package can identify all axes on /dev/input/js0, but there's no response to input.
- The joystick doesn't function on either Wayland or X11 Plasma sessions.
- No process is holding the joystick device (verified with fuser).
- Permissions are not an issue—running jstest as root still yields no input.
- Upgrading to the beta channel did not resolve the problem.

The only way to make joysticks work is to roll back to the old SteamOS version with these steps:

- Turn off the device completely.
- Press the `Power button` and `...` at the same time.
- Wait for the boot menu to appear.
- Select the previous SteamOS release using the cross buttons on the left side.
- Confirm your selection by pressing `A`

**NB:** The Steam Deck will automatically boot the latest version after a reboot. To keep using the older version, you’ll either need to avoid rebooting or perform the rollback through the boot menu each time you start the device.

**Comment by RootCellar** [comment]
the workaround posted by @JamesEnsor works on my Steam Deck. Only I would describe it as "haptic feedback" rather than a sound.

**Comment by BackAtItLikeACrackAddict** [comment]
> Same here. Haven't seen an OLED owner complaining about the controller issue.

Me i'm an oled user & i'm having the same problem. going back too 3.5 fixes it but every time i reboot, its at 3.6

**Comment by matte-schwartz** [comment]
In Game Mode, does going to Settings -> System -> Preview also work to fix the controller after it updates? That will load the Steam Deck client beta which did not have the issue for me.

**Comment by BackAtItLikeACrackAddict** [comment]
> In Game Mode, does going to Settings -> System -> Preview also work to fix the controller after it updates? That will load the Steam Deck client beta which did not have the issue for me.

I've only been on stable when i got this device. I know nothing about beta & preview.

**Comment by BackAtItLikeACrackAddict** [comment]
Sooo deleting Proton TKG has fixed my controls. I will do more testing & see if its legit or not.

**Comment by RootCellar** [comment]
I've just installed the latest update, from today, on my Steam Deck. It seems to fix the problem for me.

**Comment by kisak-valve** [comment]
Closing per the last comment.

**Comment by matte-schwartz** [comment]
this issue has regressed again when a third-party compatibility tool like proton-tkg is present on SteamOS Stable with both the Steam Deck Stable + Beta clients. The Deck controls get stuck in lizard mode when you boot into gamescope-session:

[systemreport-20250209184059.txt](https://github.com/user-attachments/files/18725769/systemreport-20250209184059.txt)

[logs.tar.gz](https://github.com/user-attachments/files/18725780/logs.tar.gz)

Switching to SteamOS Main seems to make both clients work as expected... maybe the newer controller firmware? Not sure yet.

**Comment by lostgoat** [comment]
Issues like these surface because the compat tool isn't compiled in a way that is portable across multiple OS versions. 

Previously on the steam side we added some safeguards to try to avoid bad ABI in these tools from hanging parts of the steam process. But that seems to have regressed.

In general, proton-tkg simplifies the proton build process signifiantly, but that is in exchange of losing ABI compatibility. This makes it a bad match for SteamOS. It works great for arch, but not for SteamOS (unless you are currently rebuilding the binary whenever you change OS versions, which I'm guessing many people aren't).
